### PR TITLE
Feature: coding required for data import

### PR DESCRIPTION
**Pull request by EHJ-52n**

### CODE REVIEW

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertObservationResponseV20Decoder.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    public Set<DecoderKey> getKeys() {
+        return Collections.unmodifiableSet(DECODER_KEYS);
+    }
+
+    @Override
+    public InsertObservationResponse decode(XmlObject xmlObject) throws DecodingException {
+        LOGGER.debug("REQUESTTYPE: {}", xmlObject != null ? xmlObject.getClass() : "null recevied");
+        if (xmlObject == null || !InsertObservationResponseDocument.class.isAssignableFrom(xmlObject.getClass())) {
```
Comment (autermann) [inline xmlbeans/src/main/java/org/n52/svalbard/decode/InsertObservationResponseV20Decoder.java]: Whats wrong with `xmlObject instanceof InsertObservationResponseDocument` (which also checks that `xmlObject` is not `null`)?

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultResponseDecoder.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Override
+    public Set<DecoderKey> getKeys() {
+        return Collections.unmodifiableSet(DECODER_KEYS);
+    }
+
+    @Override
+    public InsertResultResponse decode(XmlObject xmlObject) throws DecodingException {
+        LOGGER.debug("REQUESTTYPE: {}", xmlObject != null ? xmlObject.getClass() : "null recevied");
+        if (xmlObject == null || !InsertResultResponseDocument.class.isAssignableFrom(xmlObject.getClass())) {
```
Comment (autermann) [inline xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultResponseDecoder.java]: Same here...

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoder.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Override
+    public Set<DecoderKey> getKeys() {
+        return Collections.unmodifiableSet(DECODER_KEYS);
+    }
+
+    @Override
+    public InsertResultTemplateResponse decode(XmlObject xmlObject) throws DecodingException {
+        LOGGER.debug("REQUESTTYPE: {}", xmlObject != null ? xmlObject.getClass() : "null recevied");
+        if (xmlObject == null || !InsertResultTemplateResponseDocument.class.isAssignableFrom(xmlObject.getClass())) {
```
Comment (autermann) [inline xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoder.java]: Same here...

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertSensorResponseDecoder.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Override
+    public Set<DecoderKey> getKeys() {
+        return Collections.unmodifiableSet(DECODER_KEYS);
+    }
+
+    @Override
+    public InsertSensorResponse decode(XmlObject xmlObject) throws DecodingException {
+        LOGGER.debug("REQUESTTYPE: {}", xmlObject != null ? xmlObject.getClass() : "null recevied");
+        if (xmlObject == null || !InsertSensorResponseDocument.class.isAssignableFrom(xmlObject.getClass())) {
```
Comment (autermann) [inline xmlbeans/src/main/java/org/n52/svalbard/decode/InsertSensorResponseDecoder.java]: Same here...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+public class InsertObservationResponseV20DecoderTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldThrowExceptionOnWrongInputType() throws DecodingException {
+        thrown.expect(UnsupportedDecoderInputException.class);
+        thrown.expectMessage(
+                "Decoder InsertObservationResponseV20Decoder can not decode 'org.apache.xmlbeans.impl.values.XmlAnyTypeImpl'");
```
Comment (autermann) [inline xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java]: Don't require error message that specific... I mean we're using only XmlBeans interfaces, but here you require `XmlObject.Factory.newInstance()` to return a class of type `org.apache.xmlbeans.impl.values.XmlAnyTypeImpl`...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        thrown.expectMessage(
+                "Decoder InsertObservationResponseV20Decoder can not decode 'null'");
+
+        new InsertObservationResponseV20Decoder().decode(null);
+    }
+
+    @Test
+    public void shouldThrowExceptionOnMissingTypeInDocument() throws DecodingException {
+        thrown.expect(DecodingException.class);
+        thrown.expectMessage("Received XML document is not valid. Set log level to debug to get more details");
```
Comment (autermann) [inline xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java]: That's the error message? Is it actually true?

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultResponseDecoderTest.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+public class InsertResultResponseDecoderTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldThrowExceptionOnWrongInputType() throws DecodingException {
+        thrown.expect(UnsupportedDecoderInputException.class);
+        thrown.expectMessage(
+                "Decoder InsertResultResponseDecoder can not decode 'org.apache.xmlbeans.impl.values.XmlAnyTypeImpl'");
```
Comment (autermann) [inline xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultResponseDecoderTest.java]: Same as above...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoderTest.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+public class InsertResultTemplateResponseDecoderTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldThrowExceptionOnWrongInputType() throws DecodingException {
+        thrown.expect(UnsupportedDecoderInputException.class);
+        thrown.expectMessage(
+                "Decoder InsertResultTemplateResponseDecoder can not decode 'org.apache.xmlbeans.impl.values.XmlAnyTypeImpl'");
```
Comment (autermann) [inline xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoderTest.java]: Same as above....

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertSensorResponseDecoderTest.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+public class InsertSensorResponseDecoderTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void shouldThrowExceptionOnWrongInputType() throws DecodingException {
+        thrown.expect(UnsupportedDecoderInputException.class);
+        thrown.expectMessage(
+                "Decoder InsertSensorResponseDecoder can not decode 'org.apache.xmlbeans.impl.values.XmlAnyTypeImpl'");
```
Comment (autermann) [inline xmlbeans/src/test/java/org/n52/svalbard/decode/InsertSensorResponseDecoderTest.java]: Same as above...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java (hunk 5 above)
Comment (autermann) [reply xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java]: Also, do not rely on everybody have a rename-in-comment-and-string-feature in their IDE...

File: xmlbeans/src/test/java/org/n52/svalbard/encode/InsertSensorRequestEncoderTest.java [hunk 10]
```diff
@@ -93,47 +93,77 @@ public void prepare() {
...
+        encoder.create(null);
     }
 
     @Test
-    public void shoudlThrowExceptionWhenProcedureDescriptionFormatIsMissing() throws EncodingException {
+    public void shouldThrowExceptionIfServiceIsMissing() throws EncodingException {
         thrown.expect(UnsupportedEncoderInputException.class);
-        thrown.expectMessage("procedure description format not defined in InsertSensorRequest.");
-        new InsertSensorV2RequestEncoder().create(new InsertSensorRequest());
+        thrown.expectMessage(Is.is("Encoder " +
```
Comment (autermann) [inline xmlbeans/src/test/java/org/n52/svalbard/encode/InsertSensorRequestEncoderTest.java]: Is it `thrown.expectMessage(Is.is(text))` more readable than `thrown.expectMessage(text)`?

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertObservationResponseV20Decoder.java (hunk 1 above)
Comment (EHJ-52n) [reply xmlbeans/src/main/java/org/n52/svalbard/decode/InsertObservationResponseV20Decoder.java]: nothing.. good point.

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultResponseDecoder.java (hunk 2 above)
Comment (EHJ-52n) [reply xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultResponseDecoder.java]: Same here...

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoder.java (hunk 3 above)
Comment (EHJ-52n) [reply xmlbeans/src/main/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoder.java]: Same here...

File: xmlbeans/src/main/java/org/n52/svalbard/decode/InsertSensorResponseDecoder.java (hunk 4 above)
Comment (EHJ-52n) [reply xmlbeans/src/main/java/org/n52/svalbard/decode/InsertSensorResponseDecoder.java]: Same here...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java (hunk 5 above)
Comment (EHJ-52n) [reply xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java]: Good point.

File: xmlbeans/src/test/java/org/n52/svalbard/encode/InsertSensorRequestEncoderTest.java (hunk 10 above)
Comment (EHJ-52n) [reply xmlbeans/src/test/java/org/n52/svalbard/encode/InsertSensorRequestEncoderTest.java]: Used static imports.

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java (hunk 6 above)
Comment (EHJ-52n) [reply xmlbeans/src/test/java/org/n52/svalbard/decode/InsertObservationResponseV20DecoderTest.java]: Updated to something more specific.

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultResponseDecoderTest.java (hunk 7 above)
Comment (EHJ-52n) [reply xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultResponseDecoderTest.java]: Same as above...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoderTest.java (hunk 8 above)
Comment (EHJ-52n) [reply xmlbeans/src/test/java/org/n52/svalbard/decode/InsertResultTemplateResponseDecoderTest.java]: Same as above...

File: xmlbeans/src/test/java/org/n52/svalbard/decode/InsertSensorResponseDecoderTest.java (hunk 9 above)
Comment (EHJ-52n) [reply xmlbeans/src/test/java/org/n52/svalbard/decode/InsertSensorResponseDecoderTest.java]: Same as above...

**Comment by EHJ-52n** [comment]
@autermann 
Does the current state meet your requirements, dear code manager?

**Pull request merged**